### PR TITLE
Backend part1

### PR DESCRIPTION
**Pull request by shachar3**

### CODE REVIEW

File: app/models/cuisine.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+#  id            :integer          not null, primary key
+#  name          :string
+#  icon          :string
+#  created_at    :datetime         not null
+#  updated_at    :datetime         not null
+#  restaurant_id :integer
+#
+
+class Cuisine < ApplicationRecord
+  belongs_to :restaurant
```
Comment (kadishay) [inline app/models/cuisine.rb]: Are you sure this is the right relation?
see https://guides.rubyonrails.org/association_basics.html#the-belongs-to-association

File: app/models/restaurant.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+#  latitude          :float
+#  created_at        :datetime         not null
+#  updated_at        :datetime         not null
+#
+
+class Restaurant < ApplicationRecord
+  has_one :cuisine
+  has_many :reviews
+
+  def as_json(*)
```
Comment (kadishay) [inline app/models/restaurant.rb]: Have you considered not having this in the model?

File: db/migrate/20190724113717_remove_cuisine_from_restaurants.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class RemoveCuisineFromRestaurants < ActiveRecord::Migration[5.1]
```
Comment (kadishay) [inline db/migrate/20190724113717_remove_cuisine_from_restaurants.rb]: I would consider to roll back and have a clean migration process (as a best practice)

File: spec/controllers/restaurants_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
```
Comment (kadishay) [inline spec/controllers/restaurants_controller_spec.rb]: I would add, even a single test, just as a ref

File: app/models/cuisine.rb (hunk 1 above)
Comment (shachar3) [reply app/models/cuisine.rb]: You are correct. fixed in the new commit

File: app/models/restaurant.rb (hunk 2 above)
Comment (shachar3) [reply app/models/restaurant.rb]: I would love to get your input on how this should be done, i fiddled with this a little and couldn't come up with something better.

File: db/migrate/20190724113717_remove_cuisine_from_restaurants.rb (hunk 3 above)
Comment (shachar3) [reply db/migrate/20190724113717_remove_cuisine_from_restaurants.rb]: done

File: spec/controllers/restaurants_controller_spec.rb (hunk 4 above)
Comment (shachar3) [reply spec/controllers/restaurants_controller_spec.rb]: added tests

**Pull request merged**